package com.anthropic.models.messages

import com.anthropic.core.jsonMapper
import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class BrowserWaitConfigTest {

    @Test
    fun create() {
        val browserWaitConfig = BrowserWaitConfig.builder().deferLoading(true).enabled(true).build()

        assertThat(browserWaitConfig.deferLoading()).contains(true)
        assertThat(browserWaitConfig.enabled()).contains(true)
    }

    @Test
    fun roundtrip() {
        val jsonMapper = jsonMapper()
        val browserWaitConfig = BrowserWaitConfig.builder().deferLoading(true).enabled(true).build()

        val roundtrippedBrowserWaitConfig =
            jsonMapper.readValue(
                jsonMapper.writeValueAsString(browserWaitConfig),
                jacksonTypeRef<BrowserWaitConfig>(),
            )

        assertThat(roundtrippedBrowserWaitConfig).isEqualTo(browserWaitConfig)
    }
}
